Crash after fresh install and addition of plugins.

Hello,

Etherpad 2.6.1 crashes after a fresh install and addition of plugins mentionned here : 

[https://github.com/ether/etherpad#suggested-plugins](url)

```
[2026-04-18T11:36:46.393] [ERROR] settings - TypeError: /opt/etherpad-lite/node_modules/ep_etherpad-lite/templates/pad.html:67
    65|           <% e.begin_block("editbarMenuLeft"); %>
    66|           <%- toolbar.menu(settings.toolbar.left, isReadOnly, 'left', 'pad') %>
 >> 67|           <% e.end_block(); %>
    68|       </ul>
    69|       <ul class="menu_right" role="toolbar">
    70|           <% e.begin_block("editbarMenuRight"); %>

Cannot read properties of undefined (reading 'indexOf')
    at Object.skip (/opt/etherpad-lite/src/plugin_packages/.versions/ep_font_color@0.0.122/index.js:7:47)
    at Object.hook_fn (/opt/etherpad-lite/src/plugin_packages/.versions/ep_plugin_helpers@0.2.5/eejs-blocks.js:6:25)
    at callHookFnSync (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:123:16)
    at <anonymous> (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:196:54)
    at Array.map (<anonymous>)
    at Object.exports.callAll (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:196:25)
    at Object.exports.end_block (/opt/etherpad-lite/src/node/eejs/index.ts:64:9)
    at eval ("/opt/etherpad-lite/node_modules/ep_etherpad-lite/templates/pad.html":62:10)
    at pad (/opt/etherpad-lite/node_modules/.pnpm/ejs@4.0.1/node_modules/ejs/lib/cjs/ejs.js:643:23)
    at Object.exports.require (/opt/etherpad-lite/src/node/eejs/index.ts:109:15)
    at <anonymous> (/opt/etherpad-lite/src/node/hooks/express/specialpages.ts:348:28)
    at Layer.handleRequest (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/layer.js:152:17)
    at next (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/route.js:157:13)
    at Route.dispatch (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/route.js:117:3)
    at handle (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/index.js:435:11)
    at Layer.handleRequest (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/layer.js:152:17)
[2026-04-18T11:36:50.678] [ERROR] settings - TypeError: /opt/etherpad-lite/node_modules/ep_etherpad-lite/templates/pad.html:67
    65|           <% e.begin_block("editbarMenuLeft"); %>
    66|           <%- toolbar.menu(settings.toolbar.left, isReadOnly, 'left', 'pad') %>
 >> 67|           <% e.end_block(); %>
    68|       </ul>
    69|       <ul class="menu_right" role="toolbar">
    70|           <% e.begin_block("editbarMenuRight"); %>

Cannot read properties of undefined (reading 'indexOf')
    at Object.skip (/opt/etherpad-lite/src/plugin_packages/.versions/ep_font_color@0.0.122/index.js:7:47)
    at Object.hook_fn (/opt/etherpad-lite/src/plugin_packages/.versions/ep_plugin_helpers@0.2.5/eejs-blocks.js:6:25)
    at callHookFnSync (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:123:16)
    at <anonymous> (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:196:54)
    at Array.map (<anonymous>)
    at Object.exports.callAll (/opt/etherpad-lite/src/static/js/pluginfw/hooks.ts:196:25)
    at Object.exports.end_block (/opt/etherpad-lite/src/node/eejs/index.ts:64:9)
    at eval ("/opt/etherpad-lite/node_modules/ep_etherpad-lite/templates/pad.html":62:10)
    at pad (/opt/etherpad-lite/node_modules/.pnpm/ejs@4.0.1/node_modules/ejs/lib/cjs/ejs.js:643:23)
    at Object.exports.require (/opt/etherpad-lite/src/node/eejs/index.ts:109:15)
    at <anonymous> (/opt/etherpad-lite/src/node/hooks/express/specialpages.ts:348:28)
    at Layer.handleRequest (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/layer.js:152:17)
    at next (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/route.js:157:13)
    at Route.dispatch (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/route.js:117:3)
    at handle (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/index.js:435:11)
    at Layer.handleRequest (/opt/etherpad-lite/node_modules/.pnpm/router@2.2.0/node_modules/router/lib/layer.js:152:17)

```

The ep_etherpad-lite plugin is installed by default.

Thank you

--- a/src/tests/backend/specs/settings.ts
+++ b/src/tests/backend/specs/settings.ts
@@ -89,4 +89,62 @@ describe(__filename, function () {
       assert.deepEqual(settings!.ep_webrtc, {"enabled": true});
     })
   })
+
+  // Regression test for https://github.com/ether/etherpad/issues/7543.
+  // Plugins (ep_font_color, ep_font_size, ep_plugin_helpers, …) consume
+  // Settings via CommonJS require(), which under tsx/ESM interop would place
+  // the default export under .default and leave top-level fields undefined.
+  // That broke template rendering with:
+  //   TypeError: Cannot read properties of undefined (reading 'indexOf')
+  // when plugins called settings.toolbar.left / etc.
+  //
+  // The CJS compat layer in Settings.ts re-exposes every top-level field on
+  // module.exports via accessor properties, so require(...).<field> resolves
+  // even though the source uses `export default`. This test asserts that
+  // contract so a future refactor can't regress it silently.
+  describe('CJS compatibility for plugin consumers', function () {
+    it('exposes top-level fields directly on require() result', function () {
+      const cjs = require('../../../node/utils/Settings');
+      // The three fields most commonly read by first-party plugins.
+      assert.notStrictEqual(cjs.toolbar, undefined,
+          'settings.toolbar must be reachable via CJS require');
+      assert.notStrictEqual(cjs.skinName, undefined,
+          'settings.skinName must be reachable via CJS require');
+      assert.notStrictEqual(cjs.padOptions, undefined,
+          'settings.padOptions must be reachable via CJS require');
+    });
+
+    it('toolbar has the shape plugins index into (left/right/timeslider)', function () {
+      const cjs = require('../../../node/utils/Settings');
+      // ep_font_color and friends JSON.stringify(settings.toolbar) then call
+      // .indexOf on the result, so the object must be present and well-formed.
+      assert.ok(cjs.toolbar && typeof cjs.toolbar === 'object');
+      assert.ok(Array.isArray(cjs.toolbar.left));
+      assert.ok(Array.isArray(cjs.toolbar.right));
+      assert.ok(Array.isArray(cjs.toolbar.timeslider));
+    });
+
+    it('does not hide the real value under a .default wrapper', function () {
+      const cjs = require('../../../node/utils/Settings');
+      // If export-default handling regresses, consumers end up seeing a
+      // {default: {...}} wrapper and .toolbar on the wrapper is undefined.
+      // Either shape is acceptable as long as .toolbar is directly present,
+      // which is what the CJS compat shim guarantees.
+      if (cjs.default != null && cjs.default.toolbar != null) {
+        assert.strictEqual(cjs.toolbar, cjs.default.toolbar,
+            'require().toolbar must be the same object as require().default.toolbar');
+      }
+    });
+
+    it('setters propagate so reloadSettings() changes are visible to plugins', function () {
+      const cjs = require('../../../node/utils/Settings');
+      const original = cjs.title;
+      try {
+        cjs.title = 'cjs-shim-test';
+        assert.strictEqual(cjs.title, 'cjs-shim-test');
+      } finally {
+        cjs.title = original;
+      }
+    });
+  });
 });

